### ISSUE TITLE
IMAP API is unable to connect to Office365's IMAP server

### ISSUE THREAD

**Issue by nisanth074** [andris9/imapapi#10]
Hi @andris9,

IMAP API is unable to connect to Office365's IMAP server.

When I add a outlook.com email address to IMAP API

```javascript
var request = require('request');
var options = {
  'method': 'POST',
  'url': 'localhost:13000/v1/accounts',
  'headers': {
    'Content-Type': 'application/json'
  },
  body: JSON.stringify({"account":"nisanth074@outlook.com","name":"nisanth074@outlook.com","imap":{"host":"outlook.office365.com","port":993,"secure":true,"useAuthServer":true},"smtp":{"host":"smtp.office365.com","port":587,"secure":true,"useAuthServer":true}})

};
request(options, function (error, response) { 
  if (error) throw new Error(error);
  console.log(response.body);
});
```

IMAP API was unable to authenticate because it couldn't find a supported authentication mechanism

![2020-05-02 at 2 12 PM](https://user-images.githubusercontent.com/1789832/80859472-f0169080-8c7e-11ea-86c6-95898035f022.png)

Outlook's IMAP server only has the `AUTH=PLAIN` and the `AUTH=XOAUTH2` capabilities

```ruby
require "net/imap"
imap = Net::IMAP.new('outlook.office365.com', 993, usessl = true, certs = nil, verify = false)
imap.capability
```

I opened a pull request that adds support for the `XOAUTH2` authentication mechanism to imapflow  
 https://github.com/andris9/imapflow/pull/17

Would you like to merge my change?

Thanks for your time.

**Comment by andris9** [comment]
I bumped imapflow version so this should be fixed 13f42f2c5aad27cb461377451932978f01086061

**Comment by nisanth074** [comment]
@andris9 Thanks! There's more to this though. outlook's OAuth access tokens are lengthy. 512 characters are insufficient. I've opened a pull to increase the accessToken's length https://github.com/andris9/imapapi/pull/11
